pkg/proto/block.go

```diff
@@ -34,23 +36,133 @@ func (nc *NxtConsensus) BinarySize() int {
 	return 8 + len(nc.GenSignature)
 }
 
+type BlockIDType byte
+
+const (
+	SignatureID BlockIDType = iota
+	DigestID
+)
+
+type BlockID struct {
+	sig    crypto.Signature
+	dig    crypto.Digest
+	idType BlockIDType
+}
+
+func NewBlockIDFromBase58(s string) (BlockID, error) {
+	dig, err := crypto.NewDigestFromBase58(s)
```

Comment:
Is it correct to try to created a digest, that is shorter, first? It seems like we can build digest from any signature...

Reply:
May be it is better to convert string to bytes and then use `NewBlockIDFromBytes`?

Reply:
It checks that decoded length is exactly 32, so it shouldn't accept signatures.

Reply:
It checks that decoded length is exactly 32, so it shouldn't accept signatures. Internally, it converts string to bytes and checks its length for both `Digest` and `Signature`.